### ISSUE TITLE
Protecting a group of routes with an auth check

### ISSUE THREAD

**Issue by nozzlegear** [molefrog/wouter#219]
Hey! I'm in the process of converting one of my apps from a very old version of React Router to this package, but I'm having trouble with protecting some routes with an auth function. 

Basically, what I want to do is something like this:

```js
<div>
	<Switch>
		  <Route path={"/auth/login"}...</Route>
		  <Route path={"/auth/logout"}...</Route>
		  <RequiresAuth>
			  <Route path={"/home"}>...</Route>
			  <Route path={"/products"}>...</Route>
			  ...
		  </RequiresAuth>
		  <Route path={"/something-else">...</Route>
		  <Route>{"404 page not found"}</Route>
	</Switch>
</div>
```

The `RequiresAuth` component is really simple, it uses the location hook and redirects to the login page if the user isn't authenticated:

```js
function RequiresAuth(props) {
	const [location, setLocation] = useLocation();
	
	React.useEffect(() => {
		if (!userIsAuthenticated) {
			setLocation("/auth/login", { replace: true });
		}
	});
	
	return props.children;
}
```

The problem I'm running into is that the `<Switch>` seems to get confused and my default "404" route will never render when I navigate to a path that isn't covered by the routes. In fact, it won't render any route that comes after the `<RequireAuth>` component, ~~instead it will pick one of the routes inside the `<RequiresAuth>` function and redirect to it~~ (edit: the redirect turned out to be something unrelated). If I'm on a route that isn't inside the auth component or doesn't come before it, I get a blank page. 

TLDR, I can't get my default catch-all route to render. I'm also not sure if I'm protecting my routes correctly, is there a preferred way to prevent users from entering a group of routes if they don't pass an arbitrary check? I'm probably overthinking this.
